reana_client/utils.py

```diff
@@ -200,15 +203,40 @@ def _validate_environment(reana_yaml):
         workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
         _validate_serial_workflow_environment(workflow_steps)
     elif workflow_type == "yadage":
-        logging.warning(
-            "The environment validation is not implemented for Yadage workflows yet."
-        )
+        workflow_steps = reana_yaml["workflow"]["specification"]["stages"]
+        _validate_yadage_workflow_environment(workflow_steps)
     elif workflow_type == "cwl":
         logging.warning(
             "The environment validation is not implemented for CWL workflows yet."
         )
 
 
+def _validate_yadage_workflow_environment(workflow_steps):
+    """Validate environments in REANA yadage workflow.
+
+    :param workflow_steps: List of dictionaries which represents different steps involved in workflow.
+    :raises Warning: Warns user if the workflow environment is invalid in yadage workflow steps.
+    """
+    for stage in workflow_steps:
+        environment = stage["scheduler"]["step"]["environment"]
```

Comment:
I haven't tested, but this probably wouldn't work with `reana-demo-bsm-search` example which have nested structure I showed this morning. Not all `scheduler` objects have `step`, and not all `step` objects have `environment` value. So perhaps you could use `.get("step", {}).get("environment", {})` technique here for safety.

Also, since there could be nested workflow structures, steps inside steps like in this [example](https://github.com/reanahub/reana-demo-bsm-search/blob/master/workflow/databkgmc.yml#L20), we should probably parse it recursively. I'm currently working on this for parameters parsing, so perhaps this PR could be merged as is, and separate issue can be created to later address it once the implementation for parameters would be there?

Reply:
Isn't it a way to load all the nested structured and have a _final_ dict with everything loaded? Instead of parsing it ourselves recursively. In the end, Yadage has to load these nested structures sooner or later so there might some utils for this?

Reply:
That would be great to have, but I'm not familiar with such a way. Currently `yadageschemas` returns only the nested structure for us